### PR TITLE
Using LittleFS and improving `semver.c` to fix some compiler's warning

### PR DESCRIPTION
**Pull request by tuan-karma**
Dear chrisjoyce911: Thank you for your useful esp32FOTA library. I would like to added some minor improvements and bug fixes as following:

- [x] Using LitleFS instead of (deprecated) [SPIFFS](https://arduino-esp8266.readthedocs.io/en/latest/filesystem.html#spiffs-and-littlefs)
- [x] Code improvements at `semver/semver.c` to eliminate the Compiler's warnings: e.g. `warning: passing argument 3 of 'concat_num' discards 'const' qualifier ...`

Know limitations:
- No limitations added

Tested:
- On an ESP32 dev board.

I hope and believe that the above improvements is useful for future users of this library and that change not break any API of your library. Thank you for you time!

### CODE REVIEW

**Comment by tobozo** [comment]
bump

Please have a look at #91 as it implements your semver fix along with agnostic filesystem handling.

Provided the <SPIFFS.h>, <LittleFS.h> or <SD.h> library is included **before** including <esp32fota.h>, and the chosen filesystem is properly started in your `setup()` it should behave as you expect without breaking the API.

Optional declarations eventually allow hybrid situations, like storing the signature on the SD and certificate in flash memory.

There's an [open discussion](https://github.com/tobozo/esp32FOTA/discussions/5) about this.

File: library.properties [hunk 1]
```diff
@@ -1,11 +1,11 @@
-name=esp32FOTA
-version=0.1.6
+name=esp32FOTA_T
+version=0.1.0
 author=Chris Joyce
-maintainer=Chris Joyce <chris@joyce.id.au>
-sentence=A simple library for firmware OTA updates
+maintainer=Anthony <anhtuan1 at gmail dot com>
```
Comment (chrisjoyce911) [inline library.properties:4]: Add your self as a maintainer as well if you like

**Comment by tobozo** [comment]
Closing this as implemented in [0.2.0](https://github.com/chrisjoyce911/esp32FOTA/releases/tag/0.2.0).

File: library.properties [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
+sentence=A fork of Chris Joye's esp32FOTA library. Some improvements by Nguyen Anh Tuan (anhtuanb1@gmail.com)
 paragraph=Allows for firmware to be updated from a webserver, the device can check for updates at any time. Uses a simple JSON file to outline if a new firmware is avaiable.
 category=Communication
-url=https://github.com/chrisjoyce911/esp32FOTA
+url=https://github.com/tuan-karma/esp32FOTA.git
```
[the remark this answers is not in the document]
Comment (tuan-karma) [reply library.properties:8]: ok Chris

**Pull request closed without merging**